File: linked lists/234. Palindrome Linked List/isPalindrome.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def isPalindrome(self, head: Optional[ListNode]) -> bool:
        current = head
        values = []
        
        while current:
            values.append(current.val)
            current = current.next
            
        n = len(values)
        for i in range(n//2):
            if values[i] != values[n - 1 -i]:
                return False
        
        return True

File: linked lists/234. Palindrome Linked List/isPalindrome.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {boolean}
 */
var isPalindrome = function(head) {
    let current = head;
    let values = [];

    while (current) {
      values.push(current.val);
      current = current.next;
    }

    let n = values.length;
    for (let i = 0; i < Math.floor(n / 2); i++) {
      if (values[i] !== values[n - 1 - i]) {
        return false;
      }
    }

    return true;
};